src/PowerShellEditorServices/Services/Symbols/ReferenceTable.cs

```diff
@@ -19,30 +20,55 @@ internal sealed class ReferenceTable
 {
     private readonly ScriptFile _parent;
 
-    private readonly ConcurrentDictionary<string, ConcurrentBag<IScriptExtent>> _symbolReferences = new(StringComparer.OrdinalIgnoreCase);
+    private readonly ConcurrentDictionary<string, ConcurrentBag<SymbolReference>> _symbolReferences = new(StringComparer.OrdinalIgnoreCase);
 
     private bool _isInited;
 
     public ReferenceTable(ScriptFile parent) => _parent = parent;
 
     /// <summary>
-    /// Clears the reference table causing it to rescan the source AST when queried.
+    /// Clears the reference table causing it to re-scan the source AST when queried.
     /// </summary>
     public void TagAsChanged()
     {
         _symbolReferences.Clear();
         _isInited = false;
     }
 
-    // Prefer checking if the dictionary has contents to determine if initialized. The field
-    // `_isInited` is to guard against rescanning files with no command references, but will
-    // generally be less reliable of a check.
+    /// <summary>
+    /// Prefer checking if the dictionary has contents to determine if initialized. The field
+    /// `_isInited` is to guard against re-scanning files with no command references, but will
+    /// generally be less reliable of a check.
+    /// </summary>
     private bool IsInitialized => !_symbolReferences.IsEmpty || _isInited;
 
-    internal bool TryGetReferences(string command, out ConcurrentBag<IScriptExtent>? references)
+    internal IEnumerable<SymbolReference> TryGetReferences(SymbolReference? symbol)
     {
         EnsureInitialized();
-        return _symbolReferences.TryGetValue(command, out references);
+        return symbol is not null
+            && _symbolReferences.TryGetValue(symbol.SymbolName, out ConcurrentBag<SymbolReference>? bag)
+                ? bag.Where(i => SymbolTypeUtils.SymbolTypeMatches(symbol.SymbolType, i.SymbolType))
+                : Enumerable.Empty<SymbolReference>();
+    }
+
+    // Gets symbol whose name contains the position
+    internal SymbolReference? TryGetSymbolAtPosition(int line, int column) => GetAllReferences()
+        .FirstOrDefault(i => i.NameRegion.ContainsPosition(line, column));
```

Comment:
Ah so this is what you were talkin' about, I see what you mean now. Yeah ideally we're revisit this sometime after this PR because I can't imagine it scales well. We'll see if anything pops up in preview though

Reply:
Eh, it hasn't seemed bad at all. It's faster than the previous logic which was to revisit the AST, checking each (newly found and created symbol)'s extent to see if it contained the position. We cut 3/4's of that out, since now we just check essentially a cached set of symbols.